rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
Okay, I've flip flopped again, sort of:

Digging through the current RPC system, I noticed that @raphlinus has encountered a similar race situation around sending the cursor for new empty buffers; the solution there was also to schedule the idle handler.

So it is clearer to me, now, that this makes sense as a real feature of the rpc system. I'm not sure if we need a new method, that is higher priority than `schedule_idle`; in any situation where there is a backlog of active events it isn't clear that our new event should be prioritized.

I'm not going to dig into this in much more detail this weekend, but I'll explore a bit more and see if I can come up with a cleaner solution when I get a chance.